### DISCUSSION TITLE
Error encounter with op.topotools.plot_coordinates

### DISCUSSION THREAD

**Discussion by Bjonze** [PMEAL/porespy#636] [Q&A] [answered]
Dear Prof. Gostick,  
I'm very interessted in using the SNOW network extraction for a small project I'm doing in a masters course at Technical University of Denmark.
However, when trying to run the tutorial on https://porespy.org/examples/networks/tutorials/snow_basic.html i encounter an error:

**File "\anaconda3\lib\site-packages\openpnm\topotools\plottools.py", line 242, in plot_coordinates
    ax.scatter(X_temp, Y_temp, c=color, s=markersize,**

**TypeError: inner() got multiple values for argument 'ax'**

Which apparently comes from this line from plot_coordinates in the plottools.py file:

**ax.scatter(X_temp, Y_temp, c=color, s=markersize,
                   marker=marker, alpha=alpha, kwargs)**

Is there a way to circumvent or fix this error? I can see in the documentation that if ax is provided to the function plot_coordinates and plot_connections it is supposed to overlay the extracted network over the mask, however that is not what's happening right now.

I do see a previous thread from Camelot18 where you've wrote that _"trying to plot all these together in matplotlib (which is what plot_connections etc uses, is a bit more than it can handle."_ However, the binary images I'm operating on is quite small (256 x 256). Is this still an issue regarding images smaller than the 400 x 400 shown in the tutorial?

Best, 
Bjørn

**Comment by jgostick** [comment, marked as answer]
Hi Bjorn, 
These examples get tested when the docs build to confirm they all work.  I can also confirm that things work when I run the following code block:

``` python

import numpy as np
import porespy as ps
import openpnm as op
import matplotlib.pyplot as plt
ps.visualization.set_mpl_style()
np.random.seed(10)

im = ps.generators.blobs(shape=[400, 400], porosity=0.6, blobiness=2)
snow_output = ps.networks.snow2(im, voxel_size=1)

# Use this line if using OpenPNM V == 3
prj = op.io.from_porespy(snow_output.network)
# Use this line if using OpenPNM < 3
prj = op.io.PoreSpy.import_data(snow_output.network

pn = prj['net_01']
geo = prj['geo_01']

fig, ax = plt.subplots(figsize=[8, 8])
ax.imshow(im.T, cmap=plt.cm.bone);
op.topotools.plot_coordinates(ax=fig,
                              network=pn,
                              size_by=pn["pore.inscribed_diameter"],
                              color_by=pn["pore.inscribed_diameter"],
                              markersize=1000)
op.topotools.plot_connections(network=pn, ax=fig)
ax.axis("off")

```

I can only assume that your error is coming from either version conflict (we are updating openpnm to version 3) of a typo somewhere in your script.  Can you elaborate on your version of porespy and openpnm, as well as post the exact code you're running?

**Reply by Bjonze** [reply]
Thank you for the instant reply! 
I'm running the exact code above in spyder 5.1.5 through Anaconda with openpm v2.6.1, porespy v2.0.2, and matplotlib v3.5.0. 
I was thinking this might be due to a version conflict as well!

**Reply by Bjonze** [reply]
![image](https://user-images.githubusercontent.com/77355480/153104481-583a7624-71fc-4bfe-b2cb-e90b87f76698.png)

This is a screenshot of the error

**Reply by jgostick** [reply]
I'm thinking there may have been some rough edges in the plot network functions in openpnm v2.6.  Are you able to try upgrading to 2.8?

**Reply by Bjonze** [reply]
Hello Prof. Gostick, 
i managed to manually pip install openpnm==2.8.2 and it worked. Looks like v2.6 had some rough edges as you suspected. I thought i had everything up to date as I installed with **$ conda install -c conda-forge openpnm** and I did a **conda update --all** afterwards. 

![image](https://user-images.githubusercontent.com/77355480/153225496-ac64b001-7a20-4276-87a3-b9c95a93f6c1.png)

The network can be extracted and overlaid on the mask as seen above which is perfect. Thank you very much for your help and making this awesome software public.

Best, 
Bjørn
